hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/transfer/AssociateTokenRecipientsStep.java

```diff
@@ -153,4 +166,16 @@ private TokenAssociation validateAndBuildAutoAssociation(
             return null;
         }
     }
+
+    private void checkTokenAllowances(Account account, AccountID topLevelPayer, TokenID tokenId, long amount) {
+        final var tokenAllowances = account.tokenAllowancesOrElse(Collections.emptyList());
+        for (int i = 0; i < tokenAllowances.size(); i++) {
+            final var allowance = tokenAllowances.get(i);
+
+            if (topLevelPayer.equals(allowance.spenderId()) && tokenId.equals(allowance.tokenId())) {
+                final var newAllowanceAmount = allowance.amount() + amount;
+                validateTrue(newAllowanceAmount >= 0, AMOUNT_EXCEEDS_ALLOWANCE);
```

Comment:
Note that it would be just as bad for the allowance to be missing completely, as for there to be an insufficient allowance.

Reply:
Added SPENDER_DOES_NOT_HAVE_ALLOWANCE check.